### ISSUE TITLE
LESS variables are parsed as rules and not variables

### ISSUE THREAD

**Issue by daviddiasfront** [stylelint/stylelint#1833]
I'm using stylelint on LESS files, and it seems not recognize variables in LESS. I have a warning when I parse a LESS file with variables:

```less
@color-primary: #007749;
@color-primary-2: #009639;
```
My terminal prints:

```bash
Expected empty line before at-rule (at-rule-empty-line-before)
```

**Comment by m-allanson** [comment]
@daviddiasfront You'll need to tell stylelint to parse your CSS files as Less. You can do this by naming the files with`.less` or specifying the syntax in the stylelint options. See the [CSS processors section of the docs](http://stylelint.io/user-guide/css-processors/#parsing-non-standard-syntax) for more info.

If that doesn't fix the problem can you please format your issue using the GitHub issue template? Otherwise it's difficult to work out what's causing the problem.

**Comment by daviddiasfront** [comment]
Thanks @m-allanson, indeed, I forgot I need to add the postcss-less parser before putting using stylelint. Here is my code working in my environment, where I added ```{ syntax: less }``` at the end of postcss([]).

```javascript
import less from 'postcss-less';

export function lintStylesTask (done) {
  return gulp.src('./app/styles/**/*.less')
    .pipe(plumber((error) => {
      console.log(error);
      this.emit('end');
    }))
    .pipe(cache('css'))
    .pipe(postcss([
      stylelint(),
      reporter({
        clearMessages: true
      })
    ], { syntax: less }))
    .on('end', () => {
      done();
    });
}
```

Thanks again!

**Comment by jeddy3** [comment]
@daviddiasfront FYI, we [recommend](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/faq.md#how-do-i-lint-using-my-task-runner-of-choice) using [`gulp-stylelint`](https://github.com/olegskl/gulp-stylelint) for better reporting.

**Comment by daviddiasfront** [comment]
Thanks @jeddy3 for your advice, it exist any particular reason? Performance, maintainability... ?

**Comment by jeddy3** [comment]
Better reporting i.e. access to stylelint's built-in reporters which handle things like invalid options and deprecation warnings better.

**Comment by daviddiasfront** [comment]
Thank you very much @jeddy3!
